src/redecomp/partition/RCB.cpp

```diff
@@ -29,23 +29,29 @@ std::vector<EntityIndexByRank> RCB<NDIMS>::generatePartitioning(
   // Build a partitioning using recursive coordinate bisection
   auto problem_tree = BuildProblemTree( n_parts, coords_by_mesh, ghost_len );
 
+  // Ensures ranks with elements are evenly distributed throughout the partition
+  auto rank_multiplier = this->getMPIUtility().NRanks() / n_parts;
+
   for ( const auto& coords : coords_by_mesh ) {
     // Build a list of coords that belong in each of the RCB entity parts
     auto ent_idx = MPIArray<int>( &this->getMPIUtility() );
     auto ent_ghost = MPIArray<bool>( &this->getMPIUtility() );
     for ( int i{ 0 }; i < n_parts; ++i ) {
-      ent_idx[i].reserve( coords.size() );
-      ent_ghost[i].reserve( coords.size() );
+      auto rank = i * rank_multiplier;
+      ent_idx[rank].reserve( coords.size() );
+      ent_ghost[rank].reserve( coords.size() );
     }
     for ( int i{ 0 }; i < coords.size(); ++i ) {
       auto dest = DetermineDomain( problem_tree, coords[i] );
-      ent_idx[dest].push_back( i );
-      ent_ghost[dest].push_back( false );
+      auto rank = dest * rank_multiplier;
+      ent_idx[rank].push_back( i );
+      ent_ghost[rank].push_back( false );
       const auto& neighbors = problem_tree( dest ).neighbor_bboxes_;
       for ( int j{ 0 }; j < neighbors.size(); ++j ) {
         if ( problem_tree( neighbors[j] ).ghost_bbox_.contains( coords[i] ) ) {
-          ent_idx[neighbors[j]].push_back( i );
-          ent_ghost[neighbors[j]].push_back( true );
+          auto neighbor_rank = neighbors[j] * rank_multiplier;
```

Comment:
same comment about using neighbor_rank to index into the array, but is neighbor_rank a whole number?